File: backend/routes/appointments.ts
import express from 'express';
import { AppDataSource } from '../src/data-source';
import { Appointment } from '../src/entities/Appointment';
import auth from '../middleware/auth';

const router = express.Router();

// Get appointments for user
router.get('/', auth, async (req: any, res) => {
  try {
    const appointmentRepo = AppDataSource.getRepository(Appointment);
    const appointments = await appointmentRepo.find({
      where: { userId: req.user.id },
      order: { date: 'DESC' },
    });
    res.json(appointments);
  } catch (err: any) {
    res.status(500).json({ message: err.message });
  }
});

// Create appointment
router.post('/', auth, async (req: any, res) => {
  const { doctorName, specialization, date, time, notes } = req.body;
  try {
    const appointmentRepo = AppDataSource.getRepository(Appointment);
    const appointment = appointmentRepo.create({
      userId: req.user.id,
      doctorName,
      specialization,
      date: new Date(date),
      time,
      notes,
    });
    await appointmentRepo.save(appointment);
    res.json(appointment);
  } catch (err: any) {
    res.status(500).json({ message: err.message });
  }
});

// Update appointment status
router.put('/:id', auth, async (req: any, res) => {
  try {
    const appointmentRepo = AppDataSource.getRepository(Appointment);
    const appointment = await appointmentRepo.findOne({ where: { id: req.params.id } });
    if (!appointment || appointment.userId !== req.user.id)
      return res.status(404).json({ message: 'Appointment not found' });

    appointment.status = req.body.status || appointment.status;
    await appointmentRepo.save(appointment);
    res.json(appointment);
  } catch (err: any) {
    res.status(500).json({ message: err.message });
  }
});

export default router;

File: backend-python/routes/appointments.py
from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session
from datetime import datetime
from typing import List

from database.database import get_db
from models.user import User
from models.appointment import Appointment
from auth.auth import get_current_user
from schemas.appointment import AppointmentCreate, AppointmentResponse, AppointmentUpdate

router = APIRouter()

@router.get("/", response_model=List[AppointmentResponse])
async def get_appointments(
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Get appointments for user."""
    appointments = db.query(Appointment).filter(
        Appointment.user_id == current_user.id
    ).order_by(Appointment.date.desc()).all()
    
    return [
        AppointmentResponse(
            id=str(appointment.id),
            doctor_name=appointment.doctor_name,
            specialization=appointment.specialization,
            date=appointment.date,
            time=appointment.time,
            status=appointment.status,
            notes=appointment.notes,
            created_at=appointment.created_at
        )
        for appointment in appointments
    ]

@router.post("/", response_model=AppointmentResponse)
async def create_appointment(
    appointment_data: AppointmentCreate,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Create appointment."""
    db_appointment = Appointment(
        user_id=current_user.id,
        doctor_name=appointment_data.doctor_name,
        specialization=appointment_data.specialization,
        date=appointment_data.date,
        time=appointment_data.time,
        notes=appointment_data.notes,
        created_at=datetime.utcnow()
    )
    
    db.add(db_appointment)
    db.commit()
    db.refresh(db_appointment)
    
    return AppointmentResponse(
        id=str(db_appointment.id),
        doctor_name=db_appointment.doctor_name,
        specialization=db_appointment.specialization,
        date=db_appointment.date,
        time=db_appointment.time,
        status=db_appointment.status,
        notes=db_appointment.notes,
        created_at=db_appointment.created_at
    )

@router.put("/{appointment_id}", response_model=AppointmentResponse)
async def update_appointment_status(
    appointment_id: str,
    appointment_update: AppointmentUpdate,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Update appointment status."""
    appointment = db.query(Appointment).filter(
        Appointment.id == appointment_id
    ).first()
    
    if not appointment or appointment.user_id != current_user.id:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail="Appointment not found"
        )
    
    if appointment_update.status:
        appointment.status = appointment_update.status
    
    db.commit()
    db.refresh(appointment)
    
    return AppointmentResponse(
        id=str(appointment.id),
        doctor_name=appointment.doctor_name,
        specialization=appointment.specialization,
        date=appointment.date,
        time=appointment.time,
        status=appointment.status,
        notes=appointment.notes,
        created_at=appointment.created_at
    )
